win_get_url: Added force_basic_auth option

wpietruszewski opened the pull request:
##### SUMMARY
Changed way of authenticating using Basic auth to Headers per https://www.ietf.org/rfc/rfc2617.txt (Basic Authentication Scheme). I have found that current method was not working on Powershell 5.0. After some search it turned out that previous method was generating problems for number of users.


##### ISSUE TYPE
 - Bugfix Pull Request


##### COMPONENT NAME
win_get_url

##### ANSIBLE VERSION
```
ansible 2.4.1.0
  config file = /home/zew_2_12412/.ansible.cfg
  configured module search path = [u'/home/zew_2_12412/ansible/library']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```

Commit:
Changed basic auth from NetworkCredential to Base64

jborean93 commented:
Is there some sort of link or page that shows the problems with NetCredentials on Powershell 5.0? I'm curious to see what they actually are and why it might be happening.

wpietruszewski commented:
First I will mention that we are working in domain environment. Powershell 5.0.10586.117. 
First I made small test to debug the HTTP traffic
On Ansible server
`ansible -m win_get_url -a "url=http://ansible:8080/test.zip dest=C:\install\test.zip url_password=pass url_username=user" -i inventory/windows windows`

First output is stock win_get_url and there is no Authorization header.  Second is modified win_get_url with base64 method with header present. 

```
[user@ansible ~]$ nc -lk 8080
GET /test.zip HTTP/1.1
Host: ansible :8080
Connection: Keep-Alive

GET /test.zip HTTP/1.1
Authorization: Basic Nskdi43ksd
Host: ansible :8080
Connection: Keep-Alive
```
I couldn't point exact problem with a 100% certainty, all the information I could found on the internet propose different implementation of the same method. Non of which worked in my case. Site that turned me to base64 is https://stackoverflow.com/questions/12282842/how-to-login-to-website-with-basic-authentication-using-powershell.

dagwieers commented:
Beware that the deadline for getting new features/modules accepted in [Ansible v2.5 is nearing](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_5.rst), it is set to either 2018-01-15 or 2018-01-31. If you are blocked, or you need feedback, please discuss on IRC channel #ansible-windows or add a comment to [the Windows Working Group meeting agenda](https://github.com/ansible/community/issues/195) to get it resolved.

willthames commented:
This solves the problem of a lack of `get_url`'s `force_basic_auth` equivalent.

Tested and works great for my particular need (downloading a file from Bamboo, which redirects you to a login page rather than return a hard 403)

jborean93 commented:
@wpietruszewski sorry for not looking further into this sooner, are you able to change the PR slightly to add in an option `force_basic_auth` which controls whether to use your method or use the current process.

We need to be able to control whether we send the basic auth on the first request (`force_basic_auth`) or get MS to handle the auth process based on the first response.

wojciechpietruszewski-tomtom committed:
Added force_basic_auth parameter to win_get_url module

force_basic_auth provides user with option to control which mechanism to use

Updated win_get_url.py documentation section

wpietruszewski commented:
@jborean93 I've added new option that controls the the mechanism.

wojciechpietruszewski-tomtom committed:
Add missing version_added in win_get_url.py

jborean93 committed:
minor update for documentation

jborean93 commented:
Thanks @wpietruszewski for your fix for this.
